api/ccapi/include/model.h

```diff
@@ -192,6 +195,20 @@ class Model {
     const std::vector<std::string> &end_layers, ReferenceLayersType type,
     const std::vector<std::string> &type_properties = {}) = 0;
 
+  /**
+   * @brief Visit each layer inside model
+   * @param fn function to be called for each layer
+   * - call param
+   *   @param layer layer the call back is visiting
+   *   @param rc run layer context clubbed with layer, null if not initialized
+   * yet
+   *
+   */
+  virtual void
+  forEachLayer(std::function<void(Layer & /**< layer */,
+                                  nntrainer::RunLayerContext * /**< rc */)>
```

Comment:
how about `RunLayerContext &` - match the API for a layer developer.

Reply:
Yeah :) if this is after finalized, it will be Context &